.github/workflows/main-workflow.yml

```
@@ -0,0 +1,259 @@
+name: GitHub Actions Test
+on: [push]
+jobs:
+  eval_changes:
+    runs-on: ubuntu-latest  
+    name: Test changed-files
+    outputs:
+      code_changes: ${{ steps.check_changes.outputs.code_changes }}
+      doc_changes: ${{ steps.check_changes.outputs.doc_changes }}
+      nb_changes: ${{ steps.check_changes.outputs.nb_changes }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - id: check_changes
+        run: |
+          echo "=============== list modified files ==============="
+          editedFiles=`git diff --name-only HEAD^ --name-only`
```

On Azure Pipelines, the commit that is checked out is the merge commit for the PR as described [here](https://stackoverflow.com/questions/63594658/git-refs-merge-vs-head-in-pull-request), the first parent of which is the target branch (so `HEAD^` gets the correct thing to diff against).

It looks like GitHub actions doesn't do things quite the same way, so I think you'll need to use something like this instead (based on https://stackoverflow.com/questions/62331829/how-to-get-the-target-branch-of-the-github-pull-request-from-an-actions):

 ```suggestion
           editedFiles=`git diff --name-only ${{ github.event.pull_request.base.ref }}`
```